code/components/citizen-server-impl/src/state/ServerGameState.cpp

```diff
@@ -4041,20 +4114,92 @@ auto ServerGameState::GetEntityLockdownMode(const fx::ClientSharedPtr& client) -
 	return GetEntityLockdownMode();
 }
 
+
+//
+// Cleans up any outdated on-hold state bag updates
+//
+void CleanUpUnhandledStateBags(uv_timer_t* timer)
+{
+	StateBagComponent* sbac = static_cast<StateBagComponent*>(timer->data);
+	auto now = std::chrono::system_clock::now();
+	auto nextTime = now;
+
+	{
+		std::unique_lock lk(g_unhandledStateBagMutex);
+		for (auto it = g_unhandledStateBagUpdates.begin(); it != g_unhandledStateBagUpdates.end();)
+		{
+			auto& updates = it->second;
+			while (!updates.empty())
+			{
+				auto& update = updates.front();
+				if (update.time >= now)
+				{
+					if (update.time > nextTime)
+					{
+						nextTime = update.time;
+					}
+
+					break;
+				}
+
+				// remove outdated packet
+				updates.pop_front();
+			}
+
+			if (updates.empty())
+			{
+				it = g_unhandledStateBagUpdates.erase(it);
+			}
+			else
+			{
+				++it;
+			}
+		}
+	}
+
+	if (nextTime != now)
+	{
+		uv_timer_start(timer, CleanUpUnhandledStateBags, std::chrono::duration_cast<std::chrono::milliseconds>(nextTime - now).count() + 16, 0);
+	}
+}
+
 void ServerGameState::AttachToObject(fx::ServerInstanceBase* instance)
 {
 	m_instance = instance;
 
 	m_lockdownModeVar = instance->AddVariable<fx::EntityLockdownMode>("sv_entityLockdown", ConVar_None, m_entityLockdownMode, &m_entityLockdownMode);
 
 	auto sbac = fx::StateBagComponent::Create(fx::StateBagRole::Server);
+	sbac->AddSafePreCreatePrefix("entity:");
 	sbac->SetGameInterface(this);
 
+	uv_timer_init(Instance<net::UvLoopManager>::Get()->GetOrCreate("svMain")->GetLoop(), &g_unhandledStateBagTimer);
```

Comment:
Generally, using `uvw::` so we can correctly delete. In this case it has 'global' lifetime, but eh.

Reply:
I had that one set up, then I noticed that `uvw::TimerHandle` took in the data member for itself. And the custom data storage they offered was of type `std::shared_ptr<T>` so I couldn't store `fx::StateBagComponent*` without reinterpret_cast, which in turn would've broken the `std::shared_ptr` logic, sigh...

Reply:
A holding wrapper could work.